src/main/java/edu/oakland/ideas/service/IdeaDB.java

```diff
@@ -0,0 +1,269 @@
+          } else {
+            return null;
+          }
+        };
+
+    return jdbcTemplate.query(Constants.GET_WAITING_IDEAS, rowMapper);
+  }
+
+  // Ideas approved but waiting for their start_vote_date
+  public List<Idea> getFlaggedIdeas(int ideaNumber) throws DataAccessException {
+    RowMapper<Idea> rowMapper =
+        (rs, rowNum) -> {
+          if (rowNum < ideaNumber) {
+            String b = getCategoryString(rs.getInt("category"));
+            Idea flaggedIdea =
+                new Idea(
+                    rs.getInt("idea_id"),
+                    rs.getString("title"),
+                    rs.getString("description"),
+                    rs.getString("created_by"),
+                    rs.getTimestamp("created_at"),
+                    b,
+                    rs.getInt("vote_count"),
+                    0);
+            flaggedIdea.setFlagged(rs.getBoolean("is_flagged"));
+            return flaggedIdea;
+          } else {
+            return null;
+          }
+        };
+
+    return jdbcTemplate.query(Constants.GET_FLAGGED_IDEAS, rowMapper);
+  }
+
+  public List<Idea> getAdminIdeas(int category, int ideaNumber) {
+    RowMapper<Idea> rowMapper =
+        (rs, rowNum) -> {
+          if (rowNum < ideaNumber) {
+            String a = getCategoryString(rs.getInt("category"));
+            return new Idea(
+                rs.getInt("idea_id"),
+                rs.getString("title"),
+                rs.getString("description"),
+                rs.getString("created_by"),
+                rs.getTimestamp("created_at"),
+                a,
+                rs.getInt("vote_count"),
+                0);
+          } else {
+            return null;
+          }
+        };
+
+    try {
+      // TODO: Add a check for dates
+      return jdbcTemplate.query(Constants.GET_ADMIN_IDEAS, rowMapper, category);
+
+    } catch (Error e) {
+      logger.error(e);
+      return null;
+    }
+  }
+
+  public List<Idea> getArchiveIdeas(int ideaNumber) throws DataAccessException {
+    RowMapper<Idea> rowMapper =
+        (rs, rowNum) -> {
+          if (rowNum < ideaNumber) {
+            String a = getCategoryString(rs.getInt("category"));
+            return new Idea(
+                rs.getInt("idea_id"),
+                rs.getString("title"),
+                rs.getString("description"),
+                rs.getBoolean("approved"),
+                rs.getString("created_by"),
+                rs.getTimestamp("created_at"),
+                rs.getTimestamp("start_vote_date"),
+                rs.getTimestamp("end_vote_date"),
+                rs.getInt("vote_count"),
+                a);
+          } else {
+            return null;
+          }
+        };
+
+    return jdbcTemplate.query(Constants.GET_ARCHIVE_IDEAS, rowMapper);
+  }
+
+  public boolean isAdmin(String pidm) {
+    try {
+      return jdbcTemplate.queryForObject(Constants.IS_ADMIN, new Object[] {pidm}, Boolean.class);
+    } catch (Exception e) {
```

Comment:
Small thing but it might be better to catch DataAccessException here since queryForObject throws DataAccessException

Reply:
Fixed :+1: